platform/openide.loaders/src/org/openide/loaders/DataNode.java

```diff
@@ -611,6 +614,60 @@ public Long getValue() {
         
     }
     
+    private final class MIMETypeProperty extends PropertySupport.ReadOnly<String> {
+
+        public MIMETypeProperty() {
+            super("mimeType", String.class, DataObject.getString("PROP_mimeType"), DataObject.getString("HINT_mimeType"));
+        }
+
+        public String getValue() {
+            return getDataObject().getPrimaryFile().getMIMEType();
+        }
+
+    }
+    
+    private String getMIMEResolverAttribute(FileObject primaryFile, String attributeName) {
+        final String mimeType = primaryFile.getMIMEType();
+        if (mimeType != null) {
+            FileObject mimeResolversFolder = FileUtil.getConfigFile("Services/MIMEResolver"); // NOI18N
+            if (mimeResolversFolder != null) {
```

Comment:
Isn't this a bit time consuming to iterate all resolvers for every `DataNode` displayed? 

I am not sure what this method is good for, but it doesn't feel right.